### PR TITLE
feat: Support delete-resource command in cli

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <ben.ye@bytedance.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by jannfis** [comment]
Thanks @yeya24! This looks like it could use some end-to-end tests along with it, would you be willing to add some?

**Review by alexmt — APPROVED** [review]
Looks good. Thank you for the useful feature!

**Comment by alexmt** [comment]
Codegen CI task failed. You can fix it using `make clidocsgen`. I agree with @jannfis about e2e test. We need something like https://github.com/argoproj/argo-cd/blob/9559613c8f0aae0ae69488bd53d9d35524a7932c/test/e2e/app_management_test.go#L102

E.g. 


```
func TestDeleteAppResource(t *testing.T) {
	ctx := Given(t)

	ctx.
		Path(guestbookPath).
		When().
		Create().
		Sync().
		Then().
		Expect(SyncStatusIs(SyncStatusCodeInSync)).
		When().
		And(func(app *Application) {
			output, err := RunCli("app", "delete-resource", Name())
			assert.NoError(t, err)
                         // verify that resource is deleted
		}).
}
```

**Comment by yeya24** [comment]
Updated the pr. PTAL again, thank you. @alexmt @jannfis

File: test/e2e/app_management_test.go [hunk 1]
```diff
@@ -141,6 +141,25 @@ func TestAppCreation(t *testing.T) {
 		})
 }
 
+func TestDeleteAppResource(t *testing.T) {
+	ctx := Given(t)
+
+	ctx.
+		Path(guestbookPath).
+		When().
+		Create().
```
Comment (alexmt) [inline test/e2e/app_management_test.go:150]: I think the test is failing because `.Sync()` is missing.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: test/e2e/app_management_test.go (hunk 1 above)
Comment (yeya24) [reply test/e2e/app_management_test.go:150]: Thanks. Updated.

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**